content/en/blog/_posts/2021-01-25-Work-with-Persistent-Volume-and-Secrets-for-Deploying-MySQL-on-Kubernetes.md

```diff
@@ -0,0 +1,317 @@
+---
+layout: blog
+title: "Working with PersistentVolume and Secrets for Deploying MySQL on Kubernetes"
+date: 2021-01-25
+slug: work-with-persistent-volume-and-secrets-for-deploying-mysql-on-kubernetes
+---
+
+In most of full-stack applications, there is an authentication system
+for users to register their account and conduct activities in the
+system. The users’ data is usually stored in a SQL or No-SQL databases.
+To setup and deploy MySQL database on Kubernetes, you need to know 3
+major concepts: Persistent Volumes, Secrets and Persistent Volume Claim.
+In this article, we discuss them and show you how they work together to
+make the communications between database and application possible. The
+concepts covered in this article will provide you with good foundation
+for applying them to your next Kubernetes full-stack application
+deployment project.  
+
+Kubernetes Persistent Volume
+----------------------------
+
+Let’s start off by explaining what a **Persistent Volume** (**PV**) is
+and what role does it play in the [Kubernetes
+Pods](https://coding-bootcamps.com/blog/kubernetes-evolution-from-virtual-servers-and-kubernetes-architecture.html).
+In a nutshell, PV in Kubernetes are used for deploying MySQL database. A
+PV is a piece of storage in the cluster. It is a resource in the cluster
+just like a node. The Persistent volume’s lifecycle is independent from
+Pod lifecycles. It preserves data through restarting, rescheduling, and
+even deleting Pods.
+
+
+ Kubernetes Persistent Volume Claim
+-----------------------------------
+
+PVs are consumed by something called
+a **PersistentVolumeClaim** (**PVC**). Similar to a Pod, a PVC is a
+request for storage by a user. While Pods consume node resources, PVCs
+consume PV resources. Likewise, Pods can request specific levels of
+resources (CPU and Memory); whereas, PVCs can request specific size and
+access modes (e.g. read-write or read-only). In short, Pods and PV work
+hand-in-hand to use the system resources in a way that is conducive to
+their operation. 
+
+Kubernetes Secrets
+------------------
+
+Put it simply, we use [Kubernetes
+secrets](https://blockchain.dcwebmakers.com/blog/advance-topics-for-deploying-and-managing-kubernetes-containers.html)
```

Comment:
For Secret, I recommend linking to https://kubernetes.io/docs/concepts/configuration/secret/

Reply:
I did